src/include/ipc4/notification.h

```diff
@@ -168,3 +183,22 @@ static inline void ipc4_notification_watchdog_init(struct ipc4_watchdog_timeout_
 	notif->primary.r.rsp = SOF_IPC4_MESSAGE_DIR_MSG_REQUEST;
 	notif->primary.r.msg_tgt = SOF_IPC4_MESSAGE_TARGET_FW_GEN_MSG;
 }
+
+/* Input Data Payload */
+union ipc4_resource_event_data {
+	/* Raw data */
+	uint32_t dws[6];
```

Comment:
@makarukp what the answer re having a size for this data i.e. this would allow us to easily extend it as needed in the future. We have the reserved filed in teh parent structure. For backwards compat we could say size 0 means this structure is size 6, i.e size = size_of_data + 6 words.